### DISCUSSION TITLE
Texture Bitmap Scale for BSDF Blended & Roughness Alpha

### DISCUSSION THREAD

**Discussion by Disgruntoad** [mitsuba-renderer/mitsuba3#1377] [Q&A]
Hi all,

I'm working on some procedural scene generation using Mitsuba as the renderer. I've got a script to select random textures for BlendedBSDF an to set textures for RoughConductor & RoughDielectric. However, I have to provide a texture. I generated a variety of textures in the range [0,1] as exr files for the blendedbsdf. I wanted to reuse these for roughness, but the scaling should be [0, .5] or so for realistic roughness. I'm wondering if there is a parameter to easily scale the texture values? It would also be nice because exr files are incredibly large (fp32) and using a png int8 with 1/256 scaling would obvious be much preferable. Any advice // suggestions welcome.

I can regenerate the masks in [0,.5] for roughness, so this isn't really a 'new feature request' as i can achieve my goals. I'm just hoping there's an easier // more efficient way to do so.

Thanks all.

**Comment by mandyxmq** [comment]
Hi @Disgruntoad, `blendbsdf` doesn't come with a native scaling factor for roughness. If you don't want to regenerate the maps for roughness, you can load the images in python, scale their value by half, and update the roughness texture parameters. You can use `params = mi.traverse(scene)` to look for the parameters you want to update in `params`.
